Remove default Bing Maps and MapBox keys

calebmshafer opened the pull request:
Finally removing the default keys and require all apps to supply their own.

If you'd like, I can setup a new set of keys for our internal configuration of dta and ui-test-app if needed. It shouldn't be too difficult.

pmconne commented:
> If you'd like, I can setup a new set of keys for our internal configuration of dta and ui-test-app if needed.

Yes please.

pmconne commented:
display-performance-test-app will need a key too.

calebmshafer commented on test-apps/display-performance-test-app/src/frontend/DisplayPerformanceTestApp.ts:
```diff
@@ -26,6 +26,13 @@ export class DisplayPerfTestApp {
 
+    /** eslint-disable @typescript-eslint/naming-convention */
+    iModelApp.mapLayerOptions = {
+      MapBoxImagery: process.env.SVT_MAPBOX_KEY ? { key: "access_token", value: process.env.SVT_MAPBOX_KEY } : undefined,
```
@hnn0003 I noticed that you're not using env vars at all for dpta, would you want me to add this to the config.json values allowed? Or is there a better way?

pmconne replied:
@hnn0003 @DStradley probably better if everything's in the config, right? As long as you don't commit a config containing the keys.

hnn0003 replied:
Didn't see this till now. Yeah, we generally are trying not to use env variables in dpta, so it seems like having an option for this in the config json file would be better. Can we add a default value of what this would be to the dpta code, though (the actual .ts code--not a committed config file)? That way, the auto tests and any older tests would still be able to function without any retro changes?

calebmshafer replied:
No, we want to avoid publicly exposing any secrets going forward so it would always need to be configured whenever you're running the app.

pmconne replied:
I think @hnn0003 would like you to privately inform us of what keys we should be using.

DStradley replied:
@pmconne I think config will be necessary for some situations, but it seems like in others it would be useful as an environment var (set it for a machine and forget it, rather than each config).  Could we do where either works?
Also, I assume the yaml and other testing stuff where this key will have to go now is not public?

calebmshafer replied:
Okay, I'll support both and push that up in a few minutes.

The YAML pipelines will need to set these variable in a Variable Group and no they aren't exposed publicly. The builds are internal.

pmconne replied:
If env var will suffice for now, that would be preferable to making substantial changes to dpta as part of this PR.

calebmshafer replied:
Just pushed the update to the README. I'll send an email with updated keys.

calebmshafer commented:
@pmconne added the config to dta and dpta, waiting on @hnn0003 for how exactly they'd like the configuration for dpta though.

calebmshafer commented:
@pmconne the test `obscures model based on settings` is now failing on [this line](https://github.com/imodeljs/imodeljs/blob/master/full-stack-tests/core/src/frontend/standalone/BackgroundMap.test.ts#L115).

It is the 2nd test case where it fails,

```
[ { groundBias: 10, nonLocatable: true }, "model", "map", "model", "bg" ],
```

with the check from line 119.  Is this expected now that there isn't a default map?  Should I configure this test to now be a integration test so that I can use a configured set of keys.

pmconne commented:
@calebmshafer sounds like we'll have to do that.

calebmshafer commented:
@pmconne done. Switched to that being an integration test and added all the needed configs.

The new keys for DTA use are pushed to the internal source. I added you as a reviewer over there. Let me know if you see any issues.

wgoehrig commented on core/frontend/src/tile/map/MapLayerFormatRegistry.ts:
```diff
@@ -49,8 +49,8 @@ export interface MapLayerOptions {
-  constructor(opts: MapLayerOptions) {
-    this._configOptions = opts;
+  constructor(opts?: MapLayerOptions) {
+    this._configOptions = opts ? opts : {};
```
```suggestion
    this._configOptions = opts ?? {};
```

calebmshafer commented on docs/changehistory/NextVersion.md:
```diff
@@ -224,6 +224,23 @@ Upgrade instructions:
 
+## Removed default Bing Maps and MapBox keys
+
+The `@bentley/imodeljs-frontend` has always been delivered with a Bing Maps and MapBox Imagery key which should have never been publicly exposed. Both keys have now been completely removed and all applications will need to provide their own keys.
```
That sounds better to me. Thanks.
